source/common/router/shadow_writer_impl.cc

```diff
@@ -58,7 +59,9 @@ Upstream::ThreadLocalCluster* ShadowWriterImpl::getClusterAndPreprocessHeadersAn
     return nullptr;
   }
 
-  headers.setHost(shadowAppendedHost(headers.getHostValue()));
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_shadow_header_suffix")) {
```

Comment:
I'd think this can be a shadow policy config rather than a runtime guard for this per the comment from the issue.

e.g., we can add `disable_shadow_host_suffix_append` to RequestMirrorPolicy. It is false by default for backward adoption.

https://github.com/envoyproxy/envoy/blob/8879434500ba7e76ff27d41087939bfab635f358/api/envoy/config/route/v3/route_components.proto#L772

Reply:
OK, I had worked with a few people in the Slack and this was the direction that was given... I'll do the proto change instead.